app/actors/Actors.scala

```diff
@@ -11,17 +11,25 @@ import models.documentation._
 import org.slf4j.LoggerFactory
 import play.api._
 import play.api.i18n.Lang
+import play.api.i18n.MessagesApi
 import play.api.libs.concurrent.AkkaGuiceSupport
 
 class ActorsModule extends AbstractModule with AkkaGuiceSupport {
   override def configure() = {
-    bindActor[DocumentationActor]("documentation-actor")
-    bindActorFactory[DocumentationPollingActor, DocumentationPollingActor.Factory]
+    bindTypedActor(DocumentationActor, "documentation-actor")
+    bind(classOf[DocumentationPollingActor.Factory])
+      .toProvider(classOf[DocumentationPollingActorFactoryProvider])
     bind(classOf[DocumentationConfig]).toProvider(classOf[DocumentationConfigProvider])
     bind(classOf[DocumentationRedirects]).toProvider(classOf[DocumentationRedirectsProvider])
   }
 }
 
+@Singleton
+class DocumentationPollingActorFactoryProvider @Inject()(messages: MessagesApi)
+    extends Provider[DocumentationPollingActor.Factory] {
+  def get() = DocumentationPollingActor(messages, _, _)
```

Comment:
What is going on here? How does the provider obtain the two _missing_ arguments on `DocumentationPollingActor(messages, _, _)`?

Reply:
Ah, this is a provider of `Factory's with signature:

```scala
def apply(
        repos: DocumentationGitRepos,
        documentationActor: ActorRef[DocumentationActor.Command],
    ): Behavior[Tick.type]
  }
```

Got it.

Reply:
Yeah.  I agree it takes practice to read SAM conversions.  I almost switched it back, but I was already looking at all the Provider boilerplate with disgust...